[P091] Move P165 SerSwitch from playground to main repository

TD-er opened the pull request.

TD-er committed:
[P091] Move P165 SerSwitch from playground to main repository

TD-er committed:
[P091] Add documentation

Micha-he commented:
Maybe we can use the version from enesbcs from [https://github.com/enesbcs/ESPEasyPluginPlayground/blob/master/_P165_SerSwitch.ino](url)

He has integrate the MOES-WIFI-DIMMER that I use at my home.

TD-er commented:
Ah, was not aware that version was different from the playground version.

Micha-he commented:
And I didn't noticed that he didn't merge his changes to the ESPEasyPluginPlayground, otherwise I would have told you that in the forum thread.

TD-er committed:
[P091 SerSwitch] Update to use latest sources of enesbcs

Used: https://github.com/enesbcs/ESPEasyPluginPlayground/blob/master/_P165_SerSwitch.ino

TD-er committed:
Merge branch 'mega' into feature/P165

TD-er commented:
I made a test build for it: [ESPEasy_mega-20200301-15-PR_2913.zip](https://www.dropbox.com/s/jnlf432welmj43k/ESPEasy_mega-20200301-15-PR_2913.zip?dl=0)

Micha-he commented:
I'm not find any 1M-image for a test in your zip, that contain the 'Serial MCU controlled Switch'-plugin ?!?!

TD-er commented:
That's very well possible.
Looks like we no longer have a "test" 1M version anymore.
I will make a custom build for you, one moment please.

TD-er commented:
Please test this [custom_ESP8266_1M_serSwitch_p2913.bin](https://www.dropbox.com/s/vm073006uc8ocru/custom_ESP8266_1M_serSwitch_p2913.bin?dl=0)

Based on:
```
    "CONTROLLER_SET_ALL",
    "NOTIFIER_SET_NONE",
    "PLUGIN_SET_ONLY_SWITCH",
    "USES_P001",  # Switch
    "USES_P002",  # ADC
    "USES_P004",  # Dallas DS18b20
    "USES_P028",  # BME280
    "USES_P036",  # FrameOLED
    "USES_P049",  # MHZ19
    "USES_P052",  # SenseAir
    "USES_P056",  # SDS011-Dust
    "USES_P059",  # Encoder
    "USES_P081",  # Cron
    "USES_P082",  # GPS
    "USES_P091",  # SerSwitch

    "USES_C016",  # Cache Controller

    "FEATURE_MDNS",
    "FEATURE_SD",

    "USE_SETTINGS_ARCHIVE"
```

Micha-he commented:
Works here, but also with the 3s delay, which I've described in the forum.

TD-er commented:
This one can be merged?

TD-er committed:
Merge branch 'mega' into feature/P165

Micha-he commented:
Yes, no problems here. But I can't test the devices with 'real'-state response.
